### PR TITLE
AO3-5980 AO3-5623 Update admin creation script to prompt for roles

### PR DESCRIPTION
**Pull request by redsummernight**
## Issue

https://otwarchive.atlassian.net/browse/AO3-5980
https://otwarchive.atlassian.net/browse/AO3-5623

## Purpose

- Update admin creation script to prompt for roles
- Print the admin login HTTPS URL for the right environment
- Remove spaces in admin logins
- Output i18n role names in wiki output

## Testing Instructions

Someone should run the script. I did:

<details>

    Paste or enter admins, one per line, in the format

            USERNAME, EMAIL, ROLE
    or
            USERNAME, EMAIL, ROLE, ROLE, ROLE

    where USERNAME is their Org name without spaces and without the admin- prefix
    and ROLE is one of:

            communications
            docs
            open_doors
            policy_and_abuse
            superadmin
            support
            tag_wrangling
            translation

    then two line breaks to end:

    Pandemonica,monika@example.com,support,tag_wrangling
    Modeus,gummymaid@example.com,tag_wrangling
    Cerberus,upthewrong3@example.com,support,docs
    Malina,scheherazade@example.com,superadmin
    Zdrada,glitchbitch@example.com,translation,superadmin
    Azazel,majorkatsuragi@example.com,open_doors,communications
    Justice,prosecutest@example.com,translation
    Lucifer,lucifer@example.com,policy_and_abuse
    Judgement,sinmachine@example.com,policy_and_abuse
    Beelzebub,crimsonqueen@example.com,
    Hell taker,chefbravo@example.com

    Copy and paste each section into a separate file and upload to the user's Vault:

    username: admin-Pandemonica
    password: fahZ1bae
    https://127.0.0.1/admin/login

    username: admin-Modeus
    password: sieFaem0
    https://127.0.0.1/admin/login

    username: admin-Cerberus
    password: Gah6ahph
    https://127.0.0.1/admin/login

    username: admin-Malina
    password: quieBoo2
    https://127.0.0.1/admin/login

    username: admin-Zdrada
    password: je8reiVo
    https://127.0.0.1/admin/login

    username: admin-Azazel
    password: OhQu4wit
    https://127.0.0.1/admin/login

    username: admin-Justice
    password: cee0Faer
    https://127.0.0.1/admin/login

    username: admin-Lucifer
    password: zedee8Ph
    https://127.0.0.1/admin/login

    username: admin-Judgement
    password: Wahv8pho
    https://127.0.0.1/admin/login

    username: admin-Beelzebub
    password: Wei9iefi
    https://127.0.0.1/admin/login

    username: admin-Helltaker
    password: aexieT4n
    https://127.0.0.1/admin/login

    Copy and paste into the wiki at https://wiki.transformativeworks.org/mediawiki/AO3_Admins:
    |-
    | 2020-07-10 || admin-Pandemonica || Support, Tag Wrangling
    |-
    | 2020-07-10 || admin-Modeus || Tag Wrangling
    |-
    | 2020-07-10 || admin-Cerberus || AO3 Docs, Support
    |-
    | 2020-07-10 || admin-Malina || Super admin
    |-
    | 2020-07-10 || admin-Zdrada || Super admin, Translation
    |-
    | 2020-07-10 || admin-Azazel || Communications, Open Doors
    |-
    | 2020-07-10 || admin-Justice || Translation
    |-
    | 2020-07-10 || admin-Lucifer || Policy & Abuse
    |-
    | 2020-07-10 || admin-Judgement || Policy & Abuse
    |-
    | 2020-07-10 || admin-Beelzebub || UPDATE WITH USER COMMITTEE
    |-
    | 2020-07-10 || admin-Helltaker || UPDATE WITH USER COMMITTEE

</details>

and the admins had the correct roles:

<details>

    #<Admin id: 17, created_at: "2020-07-10 05:04:50", updated_at: "2020-07-10 05:04:50", email: "monika@example.com", login: "admin-Pandemonica", roles: ["support", "tag_wrangling"]>
    #<Admin id: 18, created_at: "2020-07-10 05:04:52", updated_at: "2020-07-10 05:04:52", email: "gummymaid@example.com", login: "admin-Modeus", roles: ["tag_wrangling"]>
    #<Admin id: 19, created_at: "2020-07-10 05:04:53", updated_at: "2020-07-10 05:04:53", email: "upthewrong3@example.com", login: "admin-Cerberus", roles: ["support", "docs"]>
    #<Admin id: 20, created_at: "2020-07-10 05:04:55", updated_at: "2020-07-10 05:04:55", email: "scheherazade@example.com", login: "admin-Malina", roles: ["superadmin"]>
    #<Admin id: 21, created_at: "2020-07-10 05:04:57", updated_at: "2020-07-10 05:04:57", email: "glitchbitch@example.com", login: "admin-Zdrada", roles: ["translation", "superadmin"]>
    #<Admin id: 22, created_at: "2020-07-10 05:04:58", updated_at: "2020-07-10 05:04:58", email: "majorkatsuragi@example.com", login: "admin-Azazel", roles: ["open_doors", "communications"]>
    #<Admin id: 23, created_at: "2020-07-10 05:05:00", updated_at: "2020-07-10 05:05:00", email: "prosecutest@example.com", login: "admin-Justice", roles: ["translation"]>
    #<Admin id: 24, created_at: "2020-07-10 05:05:02", updated_at: "2020-07-10 05:05:02", email: "lucifer@example.com", login: "admin-Lucifer", roles: ["policy_and_abuse"]>
    #<Admin id: 25, created_at: "2020-07-10 05:05:03", updated_at: "2020-07-10 05:05:03", email: "sinmachine@example.com", login: "admin-Judgement", roles: ["policy_and_abuse"]>
    #<Admin id: 26, created_at: "2020-07-10 05:05:05", updated_at: "2020-07-10 05:05:05", email: "crimsonqueen@example.com", login: "admin-Beelzebub", roles: []>
    #<Admin id: 27, created_at: "2020-07-10 05:05:07", updated_at: "2020-07-10 05:05:07", email: "chefbravo@example.com", login: "admin-Helltaker", roles: []>

</details>

### CODE REVIEW

File: script/create_admin.rb [hunk 1]
```diff
@@ -48,5 +65,7 @@ def multi_gets(all_text = '')
 puts "\nCopy and paste into the wiki at https://wiki.transformativeworks.org/mediawiki/AO3_Admins:\n"
 
 admins.each do |admin|
-  puts "|-\n| #{Date.today.to_formatted_s("YYYY-MM-dd")} || #{admin.login} || UPDATE WITH USER COMMITTEE"
+  role_description = admin.roles.map { |r| I18n.t("activerecord.attributes.admin/role.#{r}") }.sort.join(", ")
+  role_description = "UPDATE WITH USER COMMITTEE" if role_description.blank?
+  puts "|-\n| #{Time.zone.today.to_formatted_s('YYYY-MM-dd')} || #{admin.login} || #{role_description}"
```
Comment (sarken) [inline script/create_admin.rb:70]: For curiosity's sake, whose time zone will this be: the database admin's, the server's (UTC), or the app's (U.S. Eastern)?

Comment (redsummernight) [reply script/create_admin.rb:70]: It's [the app's](https://github.com/otwcode/otwarchive/blob/1c4efeb95b4320832c404255792d504b2f551a01/config/application.rb#L49).

**Comment by redsummernight** [comment]
With the updated script, create 4 new admins:

<details>

    Paste or enter admins, one per line, in the format

      USERNAME, EMAIL, ROLE

    or

      USERNAME, EMAIL, ROLE, ROLE, ROLE

    where

    - USERNAME is their OTW name without the admin- prefix (spaces will be removed)
    - EMAIL/ROLE can be left blank to skip updating email/roles of existing admins
    - ROLE is one of:

      communications
      docs
      open_doors
      policy_and_abuse
      superadmin
      support
      tag_wrangling
      translation

    then two line breaks to end:

    Pandemonica,monika@example.com,support,tag_wrangling
    Malina,scheherazade@example.com,superadmin
    Beelzebub,crimsonqueen@example.com,
    Hell taker,chefbravo@example.com


    For new admins, copy and paste each section into a separate file and upload to the admin's Vault:

    username: admin-Pandemonica
    password: Ethu9muu
    https://127.0.0.1/admin/login"


    username: admin-Malina
    password: Shaeco7y
    https://127.0.0.1/admin/login"


    username: admin-Beelzebub
    password: Queang5m
    https://127.0.0.1/admin/login"


    username: admin-Helltaker
    password: Ap3JooCa
    https://127.0.0.1/admin/login"


    For all saved admins, copy and paste into the wiki at https://wiki.transformativeworks.org/mediawiki/AO3_Admins:
    |-
    | 2020-07-10 || admin-Pandemonica || Support, Tag Wrangling
    |-
    | 2020-07-10 || admin-Malina || Super admin
    |-
    | 2020-07-10 || admin-Beelzebub || UPDATE WITH USER COMMITTEE
    |-
    | 2020-07-10 || admin-Helltaker || UPDATE WITH USER COMMITTEE

</details>

Update the last 4 admins and create a new one at the same time:

<details>

    Paste or enter admins, one per line, in the format

      USERNAME, EMAIL, ROLE

    or

      USERNAME, EMAIL, ROLE, ROLE, ROLE

    where

    - USERNAME is their OTW name without the admin- prefix (spaces will be removed)
    - EMAIL/ROLE can be left blank to skip updating email/roles of existing admins
    - ROLE is one of:

      communications
      docs
      open_doors
      policy_and_abuse
      superadmin
      support
      tag_wrangling
      translation

    then two line breaks to end:

    Pandemonica,monika2@example.com,support,tag_wrangling
    Malina
    Beelzebub,,
    Hell taker,,support
    Justice,prosecutest@example.com,translation


    For new admins, copy and paste each section into a separate file and upload to the admin's Vault:

    username: admin-Justice
    password: hee8Pi5a
    https://127.0.0.1/admin/login"


    For all saved admins, copy and paste into the wiki at https://wiki.transformativeworks.org/mediawiki/AO3_Admins:
    |-
    | 2020-07-10 || admin-Pandemonica || Support, Tag Wrangling
    |-
    | 2020-07-10 || admin-Malina || Super admin
    |-
    | 2020-07-10 || admin-Beelzebub || UPDATE WITH USER COMMITTEE
    |-
    | 2020-07-10 || admin-Helltaker || Support
    |-
    | 2020-07-10 || admin-Justice || Translation

</details>

**Comment by redsummernight** [comment]
This can be merged with `[skip codeship tests]`, I didn't include that in the commit message.

File: script/create_admin.rb (hunk 1 above)
Comment (redsummernight) [reply script/create_admin.rb]: Also this line was broken, [`to_formatted_s`](https://api.rubyonrails.org/v5.1/classes/DateTime.html#method-i-to_formatted_s) doesn't expect format strings like that.

**Pull request merged**